data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -337,4 +348,11 @@
     public String getClusterUrl() {
         return clusterUrl;
     }
+
+    @Override
+    public void close() throws IOException {
+        if (!leaveHttpClientOpen) {
+            httpClient.close();
```

Comment:
Should we be worried about double close?

Reply:
I don't think its our problem